R/convert.R

```diff
@@ -1,14 +1,29 @@
 #' Convert an object from one type to another
 #'
 #' @description
-#' `convert()` uses double-dispatch, because conversion depends on both `from`
-#' and `to`. The dispatch is non-standard, because `to` is a class (not an
-#' object), and it does not take advantage of inheritance (because if you
-#' convert `x` to `superFoo` you shouldn't get an instance of `Foo` back).
+#' `convert(from, to)` is a built-in generic for converting an object from
+#' one type to another. It is special in three ways:
 #'
-#' `convert()` provides built-in implementations if `from` inherits from `to`.
-#' This default strips any properties that `from` possesses that `to` does not,
-#' and resets the class.
+#' * It uses double-dispatch, because conversion depends on both `from` and
+#'   `to`.
+#'
+#' * It uses non-standard dispatch because `to` is a class, not an object.
+#'
+#' * It doesn't respect inheritance because if you convert `x` to `superFoo`
+#'   you shouldn't get an instance of `Foo` back).
```

Comment:
I actually don't know what this is trying to say. How could you get an instance of `Foo` back if inheritance was being respected?

I'm sure i understood this at some point in the past, but on reread it isn't obvious to me anymore